### DISCUSSION TITLE
Possible to stop Menu from closing on MenuItem press?

### DISCUSSION THREAD

**Discussion by dougmacknz** [adobe/react-spectrum#6622] [General]
With the `react-aria-components` Menu, is it possible to intercept one of the ways in which it is triggered to close: when pressing a menu item?

I've tried using controlled open state but it doesn't give complete control because it seems `onOpenChange` still needs to be provided for the open state to change, and there's no way to avoid `onOpenChange` from triggering on MenuItem press as far as I can tell.

**Comment by snowystinger** [comment]
Multiple selection won't close https://react-spectrum.adobe.com/react-aria/Menu.html#multiple

What is it you're trying to accomplish though? Why do you want to prevent it from closing?

**Comment by dougmacknz** [comment]
It is a rare case, but say that a MenuItem opens a modal, and you want the menu to remain open for when the modal closes.

**Reply by snowystinger** [reply]
Ah, we actually recommend that the Menu close and then the dialog opens. This discussion may help you determine how to do that https://github.com/adobe/react-spectrum/discussions/6438
